src/flb_reload.c

```diff
@@ -541,3 +562,19 @@ int flb_reload(flb_ctx_t *ctx, struct flb_cf *cf_opts)
 
     return 0;
 }
+
+int flb_signal_reload(struct flb_config *config)
```

Comment:
I suggest to rename this function to follow the prefix of the file, something like: `flb_reload_signal_in_progress` (or similar)

Reply:
Technically it would then need to be called `flb_reload_signal_reload`. The fact that it sets the `service_id_restarting` property to in progress is an implementation detail and not what it is really meant to do.

In the future hopefully it can be replaced with a sane signaling mechanism that actually talks directly to the engine to ask it to reload.